atom/browser/native_window_mac.mm

```diff
@@ -95,6 +95,33 @@ - (void)windowDidResignMain:(NSNotification*)notification {
   shell_->NotifyWindowBlur();
 }
 
+- (NSSize)windowWillResize:(NSWindow *)sender toSize:(NSSize)frameSize {
+  NSSize newSize = frameSize;
+  double interiorContentAspectRatio = shell_->GetInteriorContentAspectRatio();
+
+  if (interiorContentAspectRatio > 0.0) {
+    NSRect windowFrame = [sender frame];
+    NSRect contentViewBounds = [[sender contentView] bounds];
```

Comment:
If I get a bit of spare time I will try to get you an implementation, to try out, that uses a bit more orthodox way to achieve a frameless window on Mac OS X.

The code and documentation changes that you wanted for this pull-request are now complete. We're just going to do some testing on our side to make sure everything still works. The changes should get committed to the pull-request branch in a few hours.